webviz_subsurface/plugins/_co2_leakage/_plugin.py

```diff
@@ -0,0 +1,298 @@
+        try:
+            self._ensemble_paths = webviz_settings.shared_settings["scratch_ensembles"]
+            self._surface_server = SurfaceServer.instance(app)
+            self._polygons_server = FaultPolygonsServer.instance(app)
+
+            self._map_attribute_names = init_map_attribute_names(map_attribute_names)
+            # Surfaces
+            self._ensemble_surface_providers = init_surface_providers(
+                webviz_settings, ensembles
+            )
+            # Polygons
+            self._fault_polygon_handlers = {
+                ens: FaultPolygonsHandler(
+                    self._polygons_server,
+                    self._ensemble_paths[ens],
+                    map_surface_names_to_fault_polygons or {},
+                    fault_polygon_attribute,
+                )
+                for ens in ensembles
+            }
+            # Well picks
+            self._well_pick_providers = init_well_pick_providers(
+                self._ensemble_paths,
+                well_pick_relpath,
+                map_surface_names_to_well_pick_names,
+            )
+        except Exception as e:
+            self._error_message = f"Plugin initialization failed: {e}"
+
+        self.add_shared_settings_group(
+            ViewSettings(
+                self._ensemble_paths,
+                self._ensemble_surface_providers,
+                self._map_attribute_names,
+            ),
+            self.Ids.MAIN_SETTINGS
+        )
+        self.add_view(MainView(), self.Ids.MAIN_VIEW)
+        self.add_store(self.Ids.DATE_STORE, WebvizPluginABC.StorageType.SESSION)
+
+    @property
+    def layout(self):
+        return _error.error(self._error_message)
+
+    def _view_component(self, component_id):
+        return (
+            self.view(self.Ids.MAIN_VIEW)
+            .view_element(MainView.Ids.MAIN_ELEMENT)
+            .component_unique_id(component_id)
+            .to_string()
+        )
+
+    def _settings_component(self, component_id):
+        return (
+            self.shared_settings_group(self.Ids.MAIN_SETTINGS)
+            .component_unique_id(component_id)
+            .to_string()
+        )
+
+    def _set_callbacks(self) -> None:
+        @callback(
+            Output(self._view_component(MapViewElement.Ids.BAR_PLOT), "figure"),
+            Output(self._view_component(MapViewElement.Ids.TIME_PLOT), "figure"),
+            Input(self._settings_component(ViewSettings.Ids.ENSEMBLE), "value"),
+        )
+        def update_graphs(ensemble):
+            rz_paths = fmu_realization_paths(self._ensemble_paths[ensemble])
+            fig_args = (
+                rz_paths,
+                ENSEMBLE_PLOT_HEIGHT,
+                ENSEMBLE_PLOT_WIDTH,
+                self._co2_containment_relpath,
+            )
+            fig0 = generate_co2_volume_figure(*fig_args)
+            fig1 = generate_co2_time_containment_figure(*fig_args)
+            return fig0, fig1
+
+        @callback(
```

Comment:
Although it works, Dash is complaining about this callback.
```
A nonexistent object was used in an `Output` of a Dash callback. The id of this object is `CO2Leakage-1-main-view-main-element-date-slider` and the property is `marks`. The string ids in the current layout are: [location, webviz-content-manager, main-menu, settings-drawer, plugins-wrapper, store-date-store-CO2Leakage-1, CO2Leakage-1, CO2Leakage-1-main-settings, CO2Leakage-1-main-settings-ensemble, CO2Leakage-1-main-settings-realization, CO2Leakage-1-main-settings-formation, CO2Leakage-1-main-settings-property, CO2Leakage-1-main-settings-statistic, CO2Leakage-1-main-settings-color-scale, CO2Leakage-1-main-settings-cm-min, CO2Leakage-1-main-settings-cm-min-auto, CO2Leakage-1-main-settings-cm-max, CO2Leakage-1-main-settings-cm-max-auto, CO2Leakage-1-main-settings-plume-threshold, CO2Leakage-1-main-settings-plume-smoothing]
```
I'm not entirely sure why, but I think it is related to the use of a store, and that the callbacks are registered on the main plugin before all components are added? 
cc: @rubenthoms

Reply:
Seems probably. Found an approach to get rid of the data store, and this seems to have fixed the issue.